From 106319a7374318370815febfaeab76e8372034ad Mon Sep 17 00:00:00 2001
From: Alex Kiernan <alex.kiernan@hivehome.com>
Date: Thu, 18 Jan 2018 21:53:42 +0000
Subject: [PATCH] ostree-prepare-root: Fix split build/src compilation

---
 Makefile-switchroot.am | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile-switchroot.am b/Makefile-switchroot.am
index 70aa1c8..b1ac1a6 100644
--- a/Makefile-switchroot.am
+++ b/Makefile-switchroot.am
@@ -43,7 +43,7 @@ if BUILDOPT_USE_STATIC_COMPILER
 ostree_boot_SCRIPTS = ostree-prepare-root
 
 ostree-prepare-root : $(ostree_prepare_root_SOURCES)
-	$(STATIC_COMPILER) -o $@ -static $(ostree_prepare_root_SOURCES) $(AM_CPPFLAGS) $(AM_CFLAGS) $(DEFAULT_INCLUDES)
+	$(STATIC_COMPILER) -o $@ -static $(srcdir)/src/switchroot/ostree-prepare-root.c $(AM_CPPFLAGS) $(AM_CFLAGS) $(DEFAULT_INCLUDES)
 else
 ostree_boot_PROGRAMS += ostree-prepare-root
 ostree_prepare_root_CFLAGS = $(AM_CFLAGS) -Isrc/switchroot
-- 
2.7.4

